### PR TITLE
[4.0][com_messages] convert to prepared statement

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes

use prepared statement for SQL

### Testing Instructions
test com_messages

### Expected result
work as before

### Actual result
N/A

### CODE REVIEW

File: administrator/components/com_messages/Model/MessagesModel.php [hunk 1]
```diff
@@ -106,40 +107,56 @@ protected function getListQuery()
...
+		$id   = (int) $user->get('id');
 
 		// Select the required fields from the table.
 		$query->select(
 			$this->getState(
 				'list.select',
-				'a.*, ' .
-					'u.name AS user_from'
+				$db->quoteName('a') . '.*',
+				$db->quoteName('u.name', 'user_from')
```
Comment (SharkyKZ) [inline administrator/components/com_messages/Model/MessagesModel.php]: These should be put in an array. Now they're separate function arguments.

Comment (alikon) [reply administrator/components/com_messages/Model/MessagesModel.php]: oops

File: administrator/components/com_messages/Model/MessagesModel.php [hunk 2]
```diff
@@ -106,40 +107,58 @@ protected function getListQuery()
...
 		}
 
 		// Filter by search in subject or message.
 		$search = $this->getState('filter.search');
 
 		if (!empty($search))
 		{
-			$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
-			$query->where('a.subject LIKE ' . $search . ' OR a.message LIKE ' . $search);
+			$search = '%' . str_replace(' ', '%', trim($search) . '%';
```
Comment (Quy) [inline administrator/components/com_messages/Model/MessagesModel.php]:
```suggestion
			$search = '%' . str_replace(' ', '%', trim($search)) . '%';
```

File: administrator/components/com_messages/Model/MessagesModel.php [hunk 3]
```diff
@@ -106,40 +107,58 @@ protected function getListQuery()
...
-				'a.*, ' .
-					'u.name AS user_from'
+				[
+					$db->quoteName('a') . '.*',
+					$db->quoteName('u.name', 'user_from'),
+				]
 			)
 		);
-		$query->from('#__messages AS a');
+		$query->from($db->quoteName('#__messages', 'a');
```
Comment (Quy) [inline administrator/components/com_messages/Model/MessagesModel.php]:
```suggestion
		$query->from($db->quoteName('#__messages', 'a'));
```

File: administrator/components/com_messages/Model/ConfigModel.php [hunk 4]
```diff
@@ -56,13 +57,20 @@ protected function populateState()
 	 */
 	public function &getItem()
 	{
-		$item = new CMSObject;
+		$item   = new CMSObject;
+		$userid = (int) $this->getState('user.id')
```
Comment (Quy) [inline administrator/components/com_messages/Model/ConfigModel.php]:
```suggestion
		$userid = (int) $this->getState('user.id');
```

File: administrator/components/com_messages/Model/ConfigModel.php [hunk 5]
```diff
@@ -56,13 +57,20 @@ protected function populateState()
...
 
 		$db = $this->getDbo();
 		$query = $db->getQuery(true)
```
Comment (Quy) [inline administrator/components/com_messages/Model/ConfigModel.php]:
```suggestion
		$query = $db->getQuery(true);
```

File: administrator/components/com_messages/Model/MessageModel.php [hunk 6]
```diff
@@ -137,11 +138,18 @@ public function getItem($pk = null)
 					if ($replyId = $this->getState('reply.id'))
```
Comment (Quy) [inline administrator/components/com_messages/Model/MessageModel.php]:
```suggestion
					if ($replyId = (int) $this->getState('reply.id'))
```

Delete line 141.

Comment (alikon) [reply administrator/components/com_messages/Model/MessageModel.php]: done

File: administrator/components/com_messages/Table/MessageTable.php [hunk 7]
```diff
@@ -132,19 +133,17 @@ public function publish($pks = null, $state = 1, $userId = 0)
...
-		// Build the WHERE clause for the primary keys.
-		$where = $k . ' IN (' . implode(',', $pks) . ')';
-
 		// Update the publishing state for rows with the given primary keys.
-		$this->_db->setQuery(
-			'UPDATE ' . $this->_db->quoteName($this->_tbl)
-			. ' SET ' . $this->_db->quoteName('state') . ' = ' . (int) $state
-			. ' WHERE (' . $where . ')'
-		);
+
```
Comment (Quy) [inline administrator/components/com_messages/Table/MessageTable.php]:
```suggestion
```

File: administrator/components/com_messages/Model/MessagesModel.php [hunk 8]
```diff
@@ -106,40 +107,58 @@ protected function getListQuery()
...
 
 		if (!empty($search))
 		{
-			$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
-			$query->where('(a.subject LIKE ' . $search . ' OR a.message LIKE ' . $search . ')');
+			$search = '%' . str_replace(' ', '%', trim($search)) . '%';
+			$query->extendWhere(
+				'AND',
+				[
+					$db->quoteName('a.subject') . ' LIKE :subject ',
```
Comment (SharkyKZ) [inline administrator/components/com_messages/Model/MessagesModel.php]:
```suggestion
					$db->quoteName('a.subject') . ' LIKE :subject',
```

File: administrator/components/com_messages/Model/MessagesModel.php [hunk 9]
```diff
@@ -106,40 +107,58 @@ protected function getListQuery()
...
+					$db->quoteName('a.message') . ' LIKE :message ',
```
Comment (SharkyKZ) [inline administrator/components/com_messages/Model/MessagesModel.php]:
```suggestion
					$db->quoteName('a.message') . ' LIKE :message',
```

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 1cf1b15c514538416f8a9b7ff98af3b30bbdc167<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27075">issues.joomla.org/tracker/joomla-cms/27075</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 1cf1b15c514538416f8a9b7ff98af3b30bbdc167<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27075">issues.joomla.org/tracker/joomla-cms/27075</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27075">issues.joomla.org/tracker/joomla-cms/27075</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**